### ISSUE TITLE
How to obtain the path of combining two images into one image.

### ISSUE THREAD

**Issue by Baiwenming** [gomi42/ShapeConverter#5]
hello bro ,your code is helperful for me in my studying. but i have a question:  i have a AI file has two sharp, now two sharp are grouping a sharp ,in your project, import this AI file, will get two path, but i only want to get one path about the two grouping path。so, how to deal with this problem,hope you respones soon

**Comment by gomi42** [comment]
Hello,

that is indeed a well known problem and there is nothing you can do. 

The long version: The number of shapes that a file contains depend on how the designer constructs the final drawing (the combination of all shapes). In general a designer needs to build multiple shapes in order to get the final result. Usually he stops at that point with the result the ShapeConverter also "sees" all shapes.

In case the final drawing uses more than one color even the designer has no chance and the multiple shapes need to be kept. In case of a single color drawing Illustrator offers the feature to combine all shapes to a single shape (btw. a features that isn't well known to designers but it exists). I had to go back to my designer quite often and ask him to do this single click so that I get a single shape for his complex drawing.

The ShapeConverter just converts what's in the file to a different format. I thought a lot of times to integrate such kind of editing features. Combining shapes is a very complex task. That's why it isn't integrated yet.

Michael

**Comment by Baiwenming** [comment]
I am very pleased to receive your reply. Due to my recent busy work, I apologize for not replying to you in a timely manner. This way, I can understand why I did not perform the Grouping operation. But I still have a question to ask, which is that when I import an AI file with only one rectangle, there are two shapes parsed on the interface. There is no difference between these two shapes when selecting, but Preview only displays one. In other words, how can I distinguish between these two graphics? "GraphicVisual=fileParser. Parse (filename);" is the section where the AI file has only one graphic, but there are two numbers of GraphicVisual.

**Comment by gomi42** [comment]
To be honest I have no clear picture of what you see/get. My guess is you see 2 shapes in the left-hand shapes pane and a single shape in the preview.

As always I cannot exclude a software problem. But I've seen a lot of .ai files that repeat a shape. E.g. if you rectangle is filled AND has a border, you'll get 2 rectangles, one for the border and one for the inner filled area. All that depends on the designer and Illustrator.

**Comment by Baiwenming** [comment]
Yes, I discovered this situation this morning. As you mentioned, one has a border and the other has no border. Distinguish whether StrokeBrush is null in GraphicVisual objects. Thank you very much for your reply, but I am still very interested in the Grouping issue. If possible, I can try making some modifications myself to achieve this feature. My idea is to make some modifications by obtaining and parsing sequences. I hope to receive your suggestions. Wishing you a smooth life and work

**Comment by gomi42** [comment]
As already mentioned I spent a lot of time of how editing features could be incorporated into the ShapeConverter - especially combining shapes. The only place that makes sense is to work on the result of a parser which is a GraphicVisual. Only that way supports all input formats. 

But an editor is a totally different story. An editor requires a different user interface. But even only combining shapes to a single shape isn't trivial. Think of 2 shapes, a rectangle and a circle that overlap a little and find the resulting outline. What if both shapes have different colors set? What if there are 5 shapes and you want to combine only 2 of them?

In short do not plan or even intend to support editing features in the ShapeConverter.

Have you tried "Expression Blend"? Its free, it imports a lot of different formats and you can do any editing you can think of. The importer has its problems when it comes to gradients, the XMAL export was simply unusable for me. 

Btw, sometimes you get better results when you export SVG from Illustrator and convert the SVG with the ShapeConverter.
